src/main/java/hudson/plugins/gradle/BuildScanBuildWrapper.java

```diff
@@ -0,0 +1,71 @@
+package hudson.plugins.gradle;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractProject;
+import hudson.model.FreeStyleProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildWrapperDescriptor;
+import hudson.tasks.Builder;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class BuildScanBuildWrapper extends SimpleBuildWrapper {
+
+    @DataBoundConstructor
+    public BuildScanBuildWrapper() {
+    }
+
+    @Override
+    public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars initialEnvironment) throws IOException, InterruptedException {
+        // do nothing
+    }
+
+    @CheckForNull
+    @Override
+    public ConsoleLogFilter createLoggerDecorator(@Nonnull Run<?, ?> build) {
+        return new GradleConsoleLogFilter();
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
```

Comment:
No need to overwrite `getDescriptor()`.

Reply:
Ok, I wasn't sure if this was necessary. Is there a reason this is done so in `Gradle`?

Reply:
If the descriptor has persisted data itself, it needs to use `load` and `save`. This is true for `Gradle`, though the descriptor here has no data.

That is what I figured from the Javadoc of `Descriptor`:
```
 * <h2>Persistence</h2>
 * <p>
 * {@link Descriptor} can persist data just by storing them in fields.
 * However, it is the responsibility of the derived type to properly
 * invoke {@link #save()} and {@link #load()}.
```

Reply:
Ah sorry, my reply was for why `Gradle` overrides the constructor. I don't know why it overrides the getter...